xdem/spatial_tools.py

```diff
@@ -398,6 +408,149 @@ def hillshade(dem: Union[np.ndarray, np.ma.masked_array], resolution: Union[floa
     # The output is scaled by "(x + 0.6) / 1.84" to make it more similar to GDAL.
     return np.clip(255 * (shaded + 0.6) / 1.84, 0, 255).astype("float32")
 
+def get_xy_rotated(raster: gu.georaster.Raster, myang: float) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Rotate x, y axes of image to get along- and cross-track distances.
```

Comment:
This returns pixel coordinates rotated around the lower left corner, right? Where do the "cross-track distances" come in?

Reply:
It could also forego the raster class (if we want) by using `xdem.coreg._get_x_and_y_coords` on a transform and the shape of the array. I don't know if that is necessary or better; just a suggestion.

Reply:
Yes, could be a nice idea to combine both with an optional rotation argument.